src/plugins/intel_gpu/src/kernel_selector/cl_kernels/deconvolution_gpu_bfyx_opt.cl

```diff
@@ -18,6 +18,96 @@ KERNEL(deconvolution_gpu_bfyx_opt)(
     , FUSED_OPS_DECLS
 #endif
 )
+#if Y_AXIS_1D_FILTER == 1
```

Comment:
what about swapping x and y axis instead of introducing additional kernel?

Reply:
Added only the lines that needed to be changed to the existing code by macro.